src/apps/filters/h3ToLocalIj.c

```diff
@@ -47,32 +47,56 @@ void doCell(H3Index h, H3Index origin) {
     }
 }
 
-int main(int argc, char *argv[]) {
-    // check command line args
-    if (argc != 2) {
-        fprintf(stderr, "usage: %s [origin]\n", argv[0]);
-        exit(1);
-    }
+int main(int argc, char* argv[]) {
+    H3Index origin = 0;
+    H3Index index = 0;
 
-    H3Index origin;
+    Arg helpArg = {.names = {"-h", "--help"},
+                   .helpText = "Show this help message."};
+    Arg originArg = {
+        .names = {"-o", "--origin"},
+        .scanFormat = "%" PRIx64,
+        .valueName = "origin",
+        .value = &origin,
+        .required = true,
+        .helpText =
+            "Origin (anchoring index) for the local coordinate system."};
+    Arg indexArg = {.names = {"-i", "--index"},
+                    .scanFormat = "%" PRIx64,
+                    .valueName = "index",
+                    .value = &index,
+                    .helpText = "Index to convert to IJ coordinates."};
 
-    if (!sscanf(argv[1], "%" PRIx64, &origin))
-        error("origin could not be read");
+    Arg* args[] = {&helpArg, &originArg, &indexArg};
+    const int numArgs = 3;
+    const char* helpText = "Converts H3 indexes to local IJ coordinates";
 
-    if (!H3_EXPORT(h3IsValid)(origin)) error("origin is invalid");
+    if (parseArgs(argc, argv, numArgs, args, &helpArg, helpText)) {
+        return helpArg.found ? 0 : 1;
+    }
 
-    // process the indexes on stdin
-    char buff[BUFF_SIZE];
-    while (1) {
-        // get an index from stdin
-        if (!fgets(buff, BUFF_SIZE, stdin)) {
-            if (feof(stdin))
-                break;
-            else
-                error("reading H3 index from stdin");
-        }
+    if (!H3_EXPORT(h3IsValid)(origin)) {
```

Comment:
Same note - we should probably fail on invalid `index` or `stdin` input as well.